Runner/ExecutionInfoMapper.cs

```diff
@@ -0,0 +1,47 @@
+// Copyright 2015 ThoughtWorks, Inc.
+//
+// This file is part of Gauge-CSharp.
+//
+// Gauge-CSharp is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Gauge-CSharp is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Gauge-CSharp.  If not, see <http://www.gnu.org/licenses/>.
+
+using System.Linq;
+using Gauge.CSharp.Lib;
+
+namespace Gauge.CSharp.Runner {
+
+    public class ExecutionInfoMapper {
+        public ExecutionContext ExecutionInfoFrom(Messages.ExecutionInfo currentExecutionInfo) {
+            if (currentExecutionInfo == null)
+                return new ExecutionContext();
+            
+            return new ExecutionContext(SpecificationFrom(currentExecutionInfo.CurrentSpec), ScenarioFrom(currentExecutionInfo.CurrentScenario),
+                    StepFrom(currentExecutionInfo.CurrentStep));
+        }
+
+        public ExecutionContext.Specification SpecificationFrom(Messages.SpecInfo currentSpec)
+        {
+            return currentSpec != null ? new ExecutionContext.Specification(currentSpec.Name, currentSpec.FileName, currentSpec.IsFailed, currentSpec.Tags.ToArray()) : new ExecutionContext.Specification();
+        }
+
+        public ExecutionContext.Scenario ScenarioFrom(Messages.ScenarioInfo currentScenario)
+        {
+            return currentScenario != null ? new ExecutionContext.Scenario(currentScenario.Name, currentScenario.IsFailed, currentScenario.Tags.ToArray()) : new ExecutionContext.Scenario();
+        }
+
+        public ExecutionContext.StepDetails StepFrom(Messages.StepInfo currentStep)
+        {
+            return currentStep?.Step != null ? new ExecutionContext.StepDetails(currentStep.Step.ActualStepText, currentStep.IsFailed) : new ExecutionContext.StepDetails();
```

Comment:
`currentStep?.Step` requires .net 4.6 at the minimum? I think we have 4.5.2 as a minimum pre-requisite. I'd like to keep this compatible with 4.5.2, and at some point we can run a cleanup across project to use 4.6 features. What do you say?